src/tactic/lint.lean

```diff
@@ -266,7 +267,30 @@ meta def instance_priority (d : declaration) : tactic (option string) := do
     if info = binder_info.inst_implicit ∨ tp.get_app_fn.is_constant_of `out_param
     then none else some e,
   let always_applies := relevant_args.all expr.is_var ∧ relevant_args.nodup,
-  if always_applies then return $ some "" else return none
+  if always_applies then return $ some "set priority below 1000" else return none
+
+/- Note [default priority]:
+  Instances that always apply should be applied after instances that only apply in specific cases.
```

Comment:
Can you expand a little bit on what you mean by "instances that always apply"? This is not so clear as, of course, no instance can really always apply, you need some extra (typeclass) information. Do you mean that they will be something of the forme `my_class alpha` where `alpha` is some arbitrary type, and then the instance search will go on to check additional properties of `alpha`, in opposition to `my_class (my_nice_constructor alpha)` where it will fail right away if we try to apply it to `my_class beta` where `beta` is not `my_nice_constructor` of something?

Reply:
Yes, this is exactly what I mean. I will clarify it.